### ISSUE TITLE
Android implementation, the progress always fills 100%

### ISSUE THREAD

**Issue by Adipa** [manumaticx/circularprogress#6]
I tried following the sample use case scenario, it works on ios but always fills 100% on Android.
Only difference from the sample code is my index.js looks like this: 

```javascript
$.progress.animate({
    value: 75,
    duration: 500
});

Ti.API.info("FINISHED SHOWING indicator");
$.index.open();
```
Also I didn't notice it animate live (maybe it just happened too fast, but possibly related issues?)

**Comment by manumaticx** [comment]
I've just tested this again but it works fine for me. Here is my test code (essential the same):

```javascript

setTimeout(function(){
  
  $.progress.animate({
    value: 75,
    duration: 500
  });
  
}, 3000);

$.index.open();
```

The result:
![animate](https://cloud.githubusercontent.com/assets/2563642/6481260/3957dbd4-c25c-11e4-92a3-0e2f14573b5e.gif)

**Comment by Adipa** [comment]
@manumaticx  thanks for the response. 
So I just tested with your code and it works great, taking out the timeout set seems to cause it to fail. Is this kind of a common thing with using something in Titanium immediately after creating a page? Seems like my code was failing since it was trying to set an animation as soon as the screen showed up.

**Comment by manumaticx** [comment]
You are actually trying to do this **before** the screen shows up (calling `$.progress.animate()` before `$.index.open()`). But you are right - it's a bug, I've tested with your code again it seems there is a timing problem. I'll have a look.

**Comment by manumaticx** [comment]
Weird titanium bug. Don't ask! :wink: 

It should work now without the timeout

**Comment by Adipa** [comment]
Just looked at your commit it makes absolutely no sense but I'll take it. Thanks and good job!

**Comment by rondog** [comment]
a lot of titanium fixes I'll never understand

**Comment by manumaticx** [comment]
As I said... weird! It took me one hour to spot this. But it's a known issue which will be fixed in Ti 4.0
